File: net/JetBrains.Serialization/src/SetInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class SetInfo : IEncodableInfo
{
  [JsonProperty("Content")] private readonly List<IEncodableInfo> _content;

  [JsonConstructor]
  public SetInfo(List<IEncodableInfo> content)
  {
    _content = content;
  }

  public Asn1Encodable ToPrimitive()
  {
    return _content.ToPrimitiveDerSet();
  }
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/SetInfo.kt
package com.jetbrains.signatureverifier.serialization

import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.ASN1Primitive

@Serializable
data class SetInfo(
  val content: List<EncodableInfo>
) : EncodableInfo {

  override fun toPrimitive(): ASN1Primitive =
    content.toPrimitiveDLSet()
}